lib/tasks/compare_xml_collections.rake

```diff
@@ -0,0 +1,121 @@
+require 'libxml_to_hash'
+
+namespace :collection do
+  desc 'Translate a collection from native format to UMM JSON and back to native format'
+  task :loss, [:file, :format, :disp, :version] => :environment do |_task, args|
+    args.with_defaults(:version => '1.15.3')
+    args.with_defaults(:disp => 'show')
+
+    abort 'FORMAT INVALID' unless args.format.eql? ('echo10' || 'dif10' || 'iso19115')
+
+    filename = args.file.split('/')[-1]
+    puts "\nTranslating #{filename} to UMM JSON..."
+
+    native_original_xml = File.read(args.file)
+    native_original_hash = Hash.from_xml(native_original_xml)
+
+    #translate to UMM
+    umm_response = cmr_client.translate_collection(native_original_xml, "application/#{args.format}+xml", "application/vnd.nasa.cmr.umm+json;version=#{args.version}", skip_validation=true )
+    umm_json = umm_response.body.to_json
+    umm_response.success? ? puts("\nsuccessful translation to UMM") : abort("\nUMM translation failure")
+
+    # translate back to native
+    back_to_native = cmr_client.translate_collection(umm_json, "application/vnd.nasa.cmr.umm+json;version=#{args.version}", "application/#{args.format}+xml", skip_validation=true )
+    native_converted_hash = Hash.from_xml(back_to_native.body)
+    native_converted_xml = back_to_native.body
+    back_to_native.success? ? puts("successful translation to native format \n\n") : abort("Native format translation failure \n\n")
+
+    # nokogiri output
+    nokogiri_original = Nokogiri::XML(native_original_xml) { |config| config.strict.noblanks }
+    nokogiri_converted = Nokogiri::XML(native_converted_xml) { |config| config.strict.noblanks }
+
+    ignored_paths = Array.new
+
+    nokogiri_original.diff(nokogiri_converted, {:added => true, :removed => true}) do |change,node|
+      if node.parent.path.include?('[') && !ignored_paths.include?(node.parent.path.split('[')[0])
```

Comment:
I would memoize node.parent.path.split('[') given that you use it several times.  Ruby probably manages this under the hood for you, but if it doesn't, this will save a bunch of string operations.